pkg/provisioner/provisioner.go

```diff
@@ -204,6 +204,15 @@ type Provisioner interface {
 
 	// GetFirmwareComponents gets all firmware components available from a note
 	GetFirmwareComponents() (components []metal3api.FirmwareComponentStatus, err error)
+
+	// Get DataImage details
+	GetDataImageStatus() (nodeReservation string, nodeLastError string)
```

Comment:
I find this method confusing. What's the status here? What is node reservation outside of the context of Ironic? I haven't looked at the Ironic implementation yet, and I'm not sure what this all means.

Reply:
After looking at the implementation and how it's used, I'd propose
```go
IsDataImageReady() (bool, error)
```
Where the boolean is `Node.Reservation == ""` and `error` is built from `LastError` if necessary. I definitely don't like returning the error as a string.

Reply:
Makes sense, I have updated the function, will update this once vmedia.get api is available.